File: src/main/java/pl/training/module5/collapse/after/NotificationFormatter.java
package pl.training.module5.collapse.after;

import java.util.Objects;

public final class NotificationFormatter {
    public String format(String recipient, String message) {
        Objects.requireNonNull(recipient, "recipient must not be null");
        Objects.requireNonNull(message, "message must not be null");

        return "To: " + recipient + "\nMessage: " + message;
    }
}

File: typescript/src/module5/collapse/after/NotificationFormatter.ts
import { requireNonNull } from '../../../shared/requireNonNull.js';

export class NotificationFormatter {
  format(recipient: string, message: string): string {
    requireNonNull(recipient, 'recipient must not be null');
    requireNonNull(message, 'message must not be null');

    return 'To: ' + recipient + '\nMessage: ' + message;
  }
}
